### PR TITLE
Remove a few long deprecated methods and fully annotate others

### PR DESCRIPTION
**Pull request by bitwiseman**
# Description 
There were a few very old deprecated classes that were annoying me.  Like 3+ years deprecated.  

Removed in this PR: 
* `DeleteToken`
* `GHIssue.Label` 

# Before submitting a PR:
We love getting PRs, but we hate asking people for the same basic changes every time. 

- [x] Push your changes to a branch other than `master`. Create your PR from that branch.    
- [x] Add JavaDocs and other comments
- [x] Write tests that run and pass in CI. See [CONTRIBUTING.md](CONTRIBUTING.md) for details on how to capture snapshot data.
- [x] Run `mvn -D enable-ci clean install site` locally. This may reformat your code, commit those changes. If this command doesn't succeed, your change will not pass CI.

### CODE REVIEW

**Comment by bitwiseman** [comment]
@Sage-Pierce @nobe0716 
Does this look reasonable to you?

**Comment by Sage-Pierce** [comment]
LGTM - Are you planning on releasing with a major rev since this is _technically_ backward incompatible?

**Comment by bitwiseman** [comment]
@Sage-Pierce 
Good question. 
I don't plan to do a major version rev.  
I went back and reduced the breaking change to only a few classes.  
The breaks are now really edge case now.

File: src/main/java/org/kohsuke/github/GHIssue.java [hunk 1]
```diff
@@ -64,22 +64,14 @@
     @SkipFromToString
     protected String body;
     // for backward compatibility with < 1.63, this collection needs to hold instances of Label, not GHLabel
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHIssue.java]:
```suggestion
```

File: src/main/java/org/kohsuke/github/GHIssue.java [hunk 2]
```diff
@@ -63,23 +63,14 @@
     protected int comments;
     @SkipFromToString
     protected String body;
-    // for backward compatibility with < 1.63, this collection needs to hold instances of Label, not GHLabel
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHIssue.java]: v1.63 is 5 years ago.

File: src/main/java/org/kohsuke/github/GHIssue.java [hunk 3]
```diff
@@ -63,23 +63,14 @@
...
     protected GHMilestone milestone;
     protected GHUser closed_by;
     protected boolean locked;
 
-    /**
-     * The type Label.
-     *
-     * @deprecated use {@link GHLabel}
-     */
-    public static class Label extends GHLabel {
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHIssue.java]: This is publicly visible, but not returned from any properties.  It exists in one spot (above).

File: src/main/java/org/kohsuke/github/GHPerson.java [hunk 4]
```diff
@@ -115,31 +114,27 @@ protected synchronized void populate() throws IOException {
...
-            public Iterator<List<GHRepository>> iterator() {
-                final Iterator<GHRepository[]> pager;
-                try {
-                    pager = GitHubPageIterator.create(root.getClient(),
-                            GHRepository[].class,
-                            root.createRequest().withUrlPath("users", login, "repos").build(),
-                            pageSize);
-                } catch (MalformedURLException e) {
-                    throw new GHException("Unable to build GitHub API URL", e);
+        return () -> {
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHPerson.java]: Clean up to use even more of the existing code base.  Still behaves the same.

File: src/main/java/org/kohsuke/github/GHPerson.java [hunk 5]
```diff
@@ -178,22 +173,18 @@ public GHRepository getRepository(String name) throws IOException {
      * @return the gravatar id
      * @deprecated No longer available in the v3 API.
      */
+    @Deprecated
     public String getGravatarId() {
-        return gravatar_id;
+        return "";
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHPerson.java]: This always returns `""` anyway.  So, let's be explicit about it.

File: src/main/java/org/kohsuke/github/GHPerson.java [hunk 6]
```diff
@@ -178,22 +173,18 @@ public GHRepository getRepository(String name) throws IOException {
...
 
     /**
-     * Returns a string like 'https://secure.gravatar.com/avatar/0cb9832a01c22c083390f3c5dcb64105' that indicates the
-     * avatar image URL.
+     * Returns a string of the avatar image URL.
      *
      * @return the avatar url
      */
     public String getAvatarUrl() {
-        if (avatar_url != null)
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GHPerson.java]: If this is null, it is still more meaningful that the `""` returned from gravatar.

File: src/main/java/org/kohsuke/github/GitHubBuilder.java [hunk 7]
```diff
@@ -396,11 +398,7 @@ public GitHubBuilder withRateLimitChecker(@Nonnull RateLimitChecker coreRateLimi
      * @return the git hub builder
      */
     public GitHubBuilder withProxy(final Proxy p) {
-        return withConnector(new ImpatientHttpConnector(new HttpConnector() {
-            public HttpURLConnection connect(URL url) throws IOException {
-                return (HttpURLConnection) url.openConnection(p);
-            }
-        }));
+        return withConnector(new ImpatientHttpConnector(url -> (HttpURLConnection) url.openConnection(p)));
```
Comment (bitwiseman) [inline src/main/java/org/kohsuke/github/GitHubBuilder.java]: Found a way to do this without dynamic classes.

File: src/test/java/org/kohsuke/github/AppTest.java [hunk 8]
```diff
@@ -861,10 +862,27 @@ public void notifications() throws Exception {
...
+            assertThat(t.getReason(), oneOf("subscribed", "mention", "review_requested", "comment"));
+            assertThat(t.getTitle(), notNullValue());
+            assertThat(t.getLastCommentUrl(), notNullValue());
+            assertThat(t.getRepository(), notNullValue());
+            assertThat(t.getUpdatedAt(), notNullValue());
+            assertThat(t.getType(), oneOf("Issue", "PullRequest"));
+
+            // both thread an unread are included
+            // assertThat(t.getLastReadAt(), notNullValue());
+            // assertThat(t.isRead(), equalTo(true));
```
Comment (kshultzCB) [inline src/test/java/org/kohsuke/github/AppTest.java]: Are these commented out intentionally?

File: src/main/java/org/kohsuke/github/GHRepository.java [hunk 9]
```diff
@@ -217,16 +219,6 @@ public String getHttpTransportUrl() {
         return clone_url;
     }
 
-    /**
-     * Git http transport url string.
-     *
-     * @return the string
-     * @deprecated Typo of {@link #getHttpTransportUrl()}
-     */
-    public String gitHttpTransportUrl() {
```
Comment (kshultzCB) [inline src/main/java/org/kohsuke/github/GHRepository.java]: It took me a few eye blinks to see this 💯

File: src/test/java/org/kohsuke/github/AppTest.java [hunk 10]
```diff
@@ -861,10 +862,27 @@ public void notifications() throws Exception {
         for (GHThread t : gitHub.listNotifications().nonBlocking(true).read(true)) {
             if (!found) {
                 found = true;
+                // both thread an unread are included
```
Comment (kshultzCB) [inline src/test/java/org/kohsuke/github/AppTest.java]: I think this is a typo. It shows up again on 879. Probably meant to be this?
```suggestion
                // both read and unread are included
```

File: src/test/java/org/kohsuke/github/AppTest.java (hunk 8 above)
Comment (bitwiseman) [reply src/test/java/org/kohsuke/github/AppTest.java]: Yes, I meant this as a note to say, "This query doesn't return only records where `isRead()` is `true`."

**Pull request merged**